org.bardsoftware.eclipsito/src/org/bardsoftware/eclipsito/Boot.java

```diff
@@ -124,38 +124,58 @@ public static void main(String args[]) {
+            modulesFiles.addAll(resolveModulesResource(modulesResource));
           } else {
-        	modulesFile = new File(modulesDir);
+            String[] modulesDirArray = getModulesPaths(modulesDir);
+            for (String dir : modulesDirArray) {
+              File modulesFile = new File(dir);
+              assert modulesFile != null : "Failed to find plugins directory";
+              assert modulesFile.isDirectory() && modulesFile.canRead() : String.format("File %s is not a directory or is not readable", modulesFile.getAbsolutePath());
+              modulesFiles.add(modulesFile);
+            }
           }
-          assert modulesFile != null : "Failed to find plugins directory";
-          assert modulesFile.isDirectory() && modulesFile.canRead() : String.format("File %s is not a directory or is not readable", modulesFile.getAbsolutePath());
           assert application != null : "Application ID not specified";
           assert descriptorPattern != null : "Descriptor pattern not specified";
-          getInstance(implementationClass).run(application, modulesFile, descriptorPattern, argList);
+          getInstance(implementationClass).run(application, modulesFiles, descriptorPattern, argList);
         } catch(Exception e) {
             LOG.log(Level.SEVERE, e.getMessage(), e);
         }
     }
 
-    private static File resolveModulesResource(String modulesResource) {
-        URL modulesUrl = Boot.class.getResource(modulesResource);
-        if (modulesUrl == null) {
-          Boot.LOG.severe("Can't resolve plugin resource=" + modulesResource);
-          return null;
-        }
-        String path;
-        try {
-          path = URLDecoder.decode(modulesUrl.getPath(), "UTF-8");
-          return new File(path);
-        } catch (UnsupportedEncodingException e) {
-          Boot.LOG.log(Level.SEVERE, "Can't parse plugin location=" + modulesUrl, e);
-          return null;
-        }
-	}
+  private static List<File> resolveModulesResource(String modulesResource) {
+    String[] modulesDirArray = getModulesPaths(modulesResource);
+    List<File> modulesResources = new ArrayList<>();
+    for (String resourceDir : modulesDirArray) {
+
+      URL modulesUrl = Boot.class.getResource(resourceDir);
+      if (modulesUrl == null) {
+        Boot.LOG.severe("Can't resolve plugin resource=" + resourceDir);
+        continue;
+      }
+      String path;
+      try {
+        path = URLDecoder.decode(modulesUrl.getPath(), "UTF-8");
+        modulesResources.add(new File(path));
+      } catch (UnsupportedEncodingException e) {
+        Boot.LOG.log(Level.SEVERE, "Can't parse plugin location=" + modulesUrl, e);
+        continue;
+      }
+    }
+    return modulesResources;
+  }
+
+  private static String[] getModulesPaths(String modules) {
+    if (modules == null) {
+      return new String[0];
+    }
+    modules = modules.replaceAll("~", System.getProperty("user.home"));
```

Comment:
I don't think that we need to replace all. We need to replace ~ if it is the first component in the path. Otherwise ~ is a valid folder name.